### PR TITLE
Fix #163: Add 'Share to Brave' extension

### LINKED ISSUE #163: Add "Share To Brave" extension
Port existing extension + URL schemes from 1.6.

Currently the only URL scheme registered is `firefox://` and their internal URL schemes based on environment (`fennec`, `firefox-internal`, etc.)

From `browser-ios`: `BraveShareTo` extension & `ShareToBraveViewController.swift`

### PR DESCRIPTION
**Pull request by kylehickinson**
## Pull Request Checklist

- [x] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`
- [x] I have updated the *Unit Tests* to cover new or changed functionality
- [x] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

simulator screen shot - iphone 8 - 2018-08-31 at 09 25 03

## Notes for testing this patch

As mentioned by @iccub in #163, we should make sure to test for:

- Twitter links
- Links in notes.app

I will also add we should test it from Brave itself (if not visible tap the last option "more" and turn on "Open in Brave")

### CODE REVIEW

File: BraveShareTo/ShareToBraveViewController.swift [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    
+    override func configurationItems() -> [Any]! {
+        guard let inputItems = extensionContext?.inputItems as? [NSExtensionItem] else {
+            return []
+        }
+        
+        // Reduce all input items down to a single list of item providers
+        let attachments: [NSItemProvider] = inputItems
+            .compactMap { $0.attachments as? [NSItemProvider] }
+            .flatMap { $0 }
```
Comment (jhreis) [inline BraveShareTo/ShareToBraveViewController.swift:24]: This is deprecated and redundant. I think this was just copied, but should remove `flatMap`

File: Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -3377,9 +3429,15 @@
 				TargetAttributes = {
 					0A6231E22121F761007B429B = {
 						CreatedOnToolsVersion = 10.0;
+						DevelopmentTeam = 97533VPBBC;
```
Comment (jhreis) [inline Client.xcodeproj/project.pbxproj]: Remove.

File: Client.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -3377,9 +3429,15 @@
...
 						ProvisioningStyle = Automatic;
 						TestTargetID = F84B21BD1A090F8100AAB793;
 					};
+					27F443942135E11200296C58 = {
+						CreatedOnToolsVersion = 10.0;
+						DevelopmentTeam = 97533VPBBC;
```
Comment (jhreis) [inline Client.xcodeproj/project.pbxproj]: Remove

File: BraveShareTo/ShareToBraveViewController.swift [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+import UIKit
+import Social
+import MobileCoreServices
+
+class ShareToBraveViewController: SLComposeServiceViewController {
+    
+    // TODO: Separate scheme for debug builds, so it can be tested without need to uninstall production app.
+    private func urlScheme(for url: String) -> NSURL? {
+        return NSURL(string: "brave://open-url?url=\(url)")
```
Comment (jhreis) [inline BraveShareTo/ShareToBraveViewController.swift]: I _think_ we can start using the Swift's non `NS` classes now. This file uses a bunch and based on the docs, seems we can drop most of the obj-c based stuff. But could be wrong.

**Review by jhreis — CHANGES_REQUESTED** [review]

File: BraveShareTo/ShareToBraveViewController.swift (hunk 1 above)
Comment (kylehickinson) [reply BraveShareTo/ShareToBraveViewController.swift:24]: This is actually stuff I added, and `flatMap` isn't deprecated in this usage :) It was replaced with `compactMap` for the usage of mapping `[T?]` to `[T]`.

This usage of `flatMap` takes `[[T]]` and flattens it to `[T]`

Comment (jhreis) [reply BraveShareTo/ShareToBraveViewController.swift:24]: Oh interesting, didn't see it in doc reference, nice 😃

**Review by jhreis — APPROVED** [review]

**Pull request merged**